antarest/study/storage/variantstudy/model/command/xpansion_common.py

```diff
@@ -50,3 +56,38 @@ def apply_create_resource_commands(
     study_data.tree.save(data=data, url=url + [filename])
     output, _ = apply_config_create_resource_commands(filename, resource_type)
     return output
+
+
+def assert_xpansion_candidate_name_is_not_already_taken(candidates: dict[str, Any], candidate_name: str) -> None:
+    for candidate in candidates.values():
+        if candidate["name"] == candidate_name:
+            raise CandidateAlreadyExistsError(f"The candidate '{candidate_name}' already exists")
+
+
+def assert_link_profile_are_files(file_study: FileStudy, xpansion_candidate_dto: XpansionCandidateInternal) -> None:
+    existing_files = file_study.tree.get(["user", "expansion", "capa"])
+    for attr in [
+        "link_profile",
+        "already_installed_link_profile",
+        "direct_link_profile",
+        "indirect_link_profile",
+        "already_installed_direct_link_profile",
+        "already_installed_indirect_link_profile",
+    ]:
+        if link_file := getattr(xpansion_candidate_dto, attr, None):
+            if link_file not in existing_files:
+                raise XpansionFileNotFoundError(f"The '{attr}' file '{link_file}' does not exist")
+
+
+def assert_link_exist(file_study: FileStudy, xpansion_candidate_dto: XpansionCandidateInternal) -> None:
+    area1, area2 = xpansion_candidate_dto.link.split(" - ")
```

Comment:
We should not need to know in several places that the serialization format of the link is "area1 - area2":
it would be better to have a small model with area1 and area2 field, and a parser and a serializer to format it into the desired format (both in the DTO and in the business layer classes)